### PR TITLE
feat(deps)!: update`webpack-dev-server` to v5 and no longer lock the versions

### PR DESCRIPTION
**Pull request by SoonIter**
## Summary

1. update `webpack-dev-server` `ws` and related deps version
2. release the semver restrictions `^x.x.x`
3. unify the version of express and @types/ws

https://github.com/web-infra-dev/rspack/issues/7030

https://github.com/webpack/webpack-dev-server/blob/master/migration-v5.md

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/rspack-dev-server/package.json [hunk 1]
```diff
@@ -45,14 +45,14 @@
...
-    "webpack-dev-server": "4.13.1",
-    "ws": "8.8.1"
+    "chokidar": "^3.6.0",
+    "connect-history-api-fallback": "^2.0.0",
+    "express": "^4.17.3",
+    "http-proxy-middleware": "^2.0.3",
+    "mime-types": "^2.1.35",
+    "webpack-dev-middleware": "^7.1.0",
+    "webpack-dev-server": "^5.0.4",
+    "ws": "^8.16.0"
```
Comment (SoonIter) [inline packages/rspack-dev-server/package.json:55]: aligned with `webpack-dev-server`
image

**Comment by SoonIter** [comment]
😢 Type is failed because of phantom dependence

img_v3_02cn_4326167b-8eb4-45d8-a3ce-1cda6f73da4g

File: packages/rspack-dev-server/tests/__snapshots__/normalizeOptions.test.ts.snap [hunk 2]
```diff
@@ -28,7 +28,6 @@ exports[`normalize options snapshot no options 1`] = `
   "host": undefined,
   "hot": true,
   "liveReload": true,
-  "magicHtml": true,
```
Comment (SoonIter) [inline packages/rspack-dev-server/tests/__snapshots__/normalizeOptions.test.ts.snap:31]: `magicHtml` option is removed in webpack-dev-server now

**Comment by SoonIter** [comment]
I'll document this breaking change later

File: packages/rspack-dev-server/package.json [hunk 3]
```diff
@@ -46,13 +46,13 @@
...
-    "chokidar": "3.6.0",
-    "connect-history-api-fallback": "2.0.0",
-    "express": "4.19.2",
-    "http-proxy-middleware": "2.0.6",
-    "mime-types": "2.1.35",
-    "webpack-dev-middleware": "6.1.2",
-    "webpack-dev-server": "4.13.1",
+    "chokidar": "^3.6.0",
+    "connect-history-api-fallback": "^2.0.0",
+    "express": "^4.17.3",
```
Comment (chenjiahan) [inline packages/rspack-dev-server/package.json:51]: we can use the latest version

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
Added `breaking change` label, as this should be considered as a breaking change.

**Pull request merged**